common/include/pcl/pcl_macros.h

```diff
@@ -313,13 +336,14 @@ pcl_round (float number)
   #define PCL_PRAGMA
 #endif
 
-// Macro for emitting pragma warning
+// Macro for emitting pragma warning for deprecated headers
```

Comment:
Let's define these in order, first the impl,then the public macro

Reply:
Shift it as a whole or just the impl?
```
// Macro for pragma operator
#if (defined (__GNUC__) || defined(__clang__))
  #define PCL_PRAGMA(x) _Pragma (#x)
#elif _MSC_VER
  #define PCL_PRAGMA(x) __pragma (#x)
#else
  #define PCL_PRAGMA
#endif

// Macro for emitting pragma warning for deprecated headers
#if (defined (__GNUC__) || defined(__clang__))
  #define _PCL_DEPRECATED_HEADER_IMPL(Message) PCL_PRAGMA (GCC warning Message)
#elif _MSC_VER
  #define _PCL_DEPRECATED_HEADER_IMPL(Message) PCL_PRAGMA (warning (Message))
#else
  #warning "You need to implement _PCL_DEPRECATED_HEADER_IMPL for this compiler"
  #define _PCL_DEPRECATED_HEADER_IMPL(Message)
#endif
```

Reply:
Define a macro before you use it